### PR TITLE
feat: Add qctrl validation

### PR DESCRIPTION
**Pull request by eginez**
### Summary
This PR introduces changes to validate inputs for qctrl's primitives.

### Details and comments
- Relies on the `channel_strategy` property in the Account class which flags what kind of primitive should be used, as shown in https://github.com/Qiskit/qiskit-ibm-runtime/pull/1017
- Routes parameter validation to qctrl's logic if the `q-ctrl` string is found in the `channel_strategy` property
- Implements parameter validation.

### CODE REVIEW

**Comment by kt474** [comment]
The `channel_strategy` property will be a part of both the service and account classes. It might be easier to create a branch off of #1017 

```python
service = QiskitRuntimeService(channel="ibm_cloud", channel_strategy="q-ctrl")
service._channel_strategy # returns "q-ctrl"
```

Adding the validation in Estimator and Sampler where the current options are validated works for me.

File: qiskit_ibm_runtime/utils/qctrl.py
Comment (ThomasMerkh) [inline qiskit_ibm_runtime/utils/qctrl.py:10]: typo
```suggestion
    return _validate_qctrl_options(
        skip_transpilation=options.get('skip_transpilation', False),
        transpilation_settings=options.get('transpilation_settings'),
        resilience_settings=options.get('resilience_settings'))

def _validate_qctrl_options(
```

**Comment by eginez** [comment]
Great, thanks for the input @kt474, will wait until #1017 merges.

File: qiskit_ibm_runtime/utils/qctrl.py
Comment (kt474) [inline qiskit_ibm_runtime/utils/qctrl.py:86]: Related to #1048 and #1049, I'd prefer if we set the proper defaults when the Q-Ctrl strategy is selected instead of overriding them like this. I haven't been able to test the full job flow yet but the rest of PR looks good to me

File: qiskit_ibm_runtime/utils/qctrl.py
Comment (kt474) [inline qiskit_ibm_runtime/utils/qctrl.py:47]: Can this be replaced with `Options.validate_options(options)`?

File: qiskit_ibm_runtime/utils/qctrl.py
Comment (eginez) [reply qiskit_ibm_runtime/utils/qctrl.py:85]: We'll need this to be able to update settings on improper user input, even after meaningful defaults.

File: qiskit_ibm_runtime/utils/qctrl.py
Comment (kt474) [inline qiskit_ibm_runtime/utils/qctrl.py:64]: I think you mean `optimization_level` here

File: qiskit_ibm_runtime/utils/qctrl.py
Comment (kt474) [inline qiskit_ibm_runtime/utils/qctrl.py:58]:
```suggestion
        description=(
            "Q-CTRL Primitives do not support resilience level 0. Please "
            "set resilience_level to 1 and re-try"
        ),
```
minor nitpick but the error msg would print with a better format

File: qiskit_ibm_runtime/utils/qctrl.py
Comment (eginez) [reply qiskit_ibm_runtime/utils/qctrl.py]: yeap, thanks for the catch! 👁️

**Review by kt474 — APPROVED** [review]
LGTM

**Comment by eginez** [comment]
Thanks @kt474, I don't have write access, would you be able to merge this PR for me?

**Pull request merged**
